ci: do not override bazel action_env on Windows

filipesilva opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #27514


## What is the new behavior?
Fix #27514 for Windows CI.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

filipesilva committed:
ci: do not override bazel action_env on Windows

Fix #27514

mary-poppins commented:
You can preview be33e3d at https://pr30510-be33e3d.ngbuilds.io/.

mary-poppins commented:
You can preview c438052 at https://pr30510-c438052.ngbuilds.io/.

filipesilva committed:
ci: describe next steps for Bazel cache on Windows CI

filipesilva committed:
ci: add descriptive titles to Codefresh steps

alan-agius4 reviewed (approved):
LGTM.

Kindly rebase as you have a conflict.

filipesilva committed:
ci: remove msys workaround

Verified it isn't a problem anymore https://github.com/docker/for-win/issues/2920

filipesilva committed:
ci: remove buildkite setup

We don't use it anymore and it's pretty stable by now.

gkalpak commented on .codefresh/bazel.rc:
```diff
@@ -2,11 +2,17 @@
 build --repository_cache=C:/codefresh/volume/bazel_repository_cache
+# Setting the output_base to a Docker volume is currently because of a Docker bug on Windows:
+# https://github.com/moby/moby/issues/37024
+# This affects Bazel because bazel_output_base\external\bazel_tools is a absolute path junction.
```
_a_ --> _an_

filipesilva replied:
Fixed, and that also made me notice `is currently because` -> `is currently broken because` just above.
